lib/mayaUsd/ufe/UsdConnectionHandler.cpp

```diff
@@ -140,20 +177,10 @@ bool UsdConnectionHandler::createConnection(
                     || dstBaseName == UsdShadeTokens->displacement)) {
                 // Create the required output based on the type of the shader node we are trying to
                 // connect:
-                UsdShadeShader       srcShader(srcUsdAttr->usdPrim());
-                PXR_NS::SdrRegistry& registry = PXR_NS::SdrRegistry::GetInstance();
-                PXR_NS::TfToken      srcInfoId;
-                srcShader.GetIdAttr().Get(&srcInfoId);
                 PXR_NS::SdrShaderNodeConstPtr srcShaderNodeDef
-                    = registry.GetShaderNodeByIdentifier(srcInfoId);
-                if (!srcShaderNodeDef) {
-                    TF_RUNTIME_ERROR(
-                        "Could not find node definition '%s' for node '%s'.",
-                        srcInfoId.GetText(),
-                        Ufe::PathString::string(srcAttr->sceneItem()->path()).c_str());
-                    return false;
-                }
-                TfToken renderContext = srcShaderNodeDef->GetSourceType() == "glslfx"
+                    = _GetShaderNodeDef(srcUsdAttr->usdPrim(), srcBaseName);
+                TfToken renderContext
+                    = !srcShaderNodeDef || srcShaderNodeDef->GetSourceType() == "glslfx"
```

Comment:
If we do not find a shader node def, assume the user knows what he is doing and that connecting on the regular surface port is what he wants.